hs_core/tests/api/native/test_bag_ingestion.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+import os
+import tempfile
+import zipfile
+import shutil
+from unittest import TestCase
+
+from django.contrib.auth.models import Group, User
+from django.core.files.uploadedfile import UploadedFile
+from rdflib import Graph, URIRef
+from rdflib.compare import _squashed_graphs_triples
+from rdflib.namespace import DCTERMS, RDF
+
+from hs_composite_resource.models import CompositeResource
+from hs_core.hydroshare import resource, add_resource_files, current_site_url
+from hs_core.testing import MockIRODSTestCaseMixin
+from hs_core import hydroshare
+from hs_file_types.models import GenericLogicalFile, FileSetLogicalFile, GeoFeatureLogicalFile, GeoRasterLogicalFile, \
+    NetCDFLogicalFile, RefTimeseriesLogicalFile
+
+
+class TestCreateResource(MockIRODSTestCaseMixin, TestCase):
+    def setUp(self):
+        super(TestCreateResource, self).setUp()
+
+        self.tmp_dir = tempfile.mkdtemp()
+        self.hs_group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+        # create a user
+        self.user = hydroshare.create_account(
+            'test_user@email.com',
+            username='mytestuser',
+            first_name='some_first_name',
+            last_name='some_last_name',
+            superuser=False,
+            groups=[self.hs_group]
+        )
+
+        self.test_bag_path = 'hs_core/tests/data/test_resource_metadata_files.zip'
+        self.extracted_directory = "temp"
+
+        with zipfile.ZipFile(self.test_bag_path, 'r') as zip_ref:
+            zip_ref.extractall(self.extracted_directory)
+
+    def tearDown(self):
+        super(TestCreateResource, self).tearDown()
+
+        shutil.rmtree(self.tmp_dir)
+
+        self.user.uaccess.delete()
+        self.user.delete()
+        self.hs_group.delete()
+
+        User.objects.all().delete()
+        Group.objects.all().delete()
+        CompositeResource.objects.all().delete()
+
+        shutil.rmtree(self.extracted_directory)
+
+    def test_bag_ingestion(self):
+        def normalize_metadata(metadata_str):
+            """Prepares metadata string to match resource id and hydroshare url of original"""
+            return metadata_str\
+                .replace(current_site_url(), "http://www.hydroshare.org")\
+                .replace(res.short_id, "97523bdb7b174901b3fc2d89813458f1")
+
+        # create empty resource
+        res = resource.create_resource(
+            'CompositeResource',
+            self.user,
+            'My Test Resource'
+            )
+        full_paths = {}
+
+        files_to_upload = [UploadedFile(file=open(self.test_bag_path, 'rb'),
+                                        name="test_resource_metadata_files.zip")]
+        add_resource_files(res.short_id, *files_to_upload, full_paths=full_paths)
+        from hs_core.views.utils import unzip_file
```

Comment:
It'd be good to move the import to beginning of the method if it cannot be imported at the beginning due to circular imports.